### ISSUE TITLE
arcade.sprite_list.SpriteList.__iter__() has incorrect type hint

### ISSUE THREAD

**Issue by dementati** [pvcraven/arcade#694]
## Bug Report
`arcade.sprite_list.SpriteList.__iter__()` has a type hint indicating that it returns an `Iterable[_SpriteType]`. According to https://docs.python.org/3/reference/datamodel.html#object.__iter__, this method should be returning an iterator, and the concrete type returned is indeed an iterator. I. e., this type hint is too general. This can trigger unnecessary mypy errors. 

### Actual behavior:
```
import arcade
foo: arcade.SpriteList = arcade.SpriteList()
for s in foo:
    pass
```

Running mypy on the example code above gives: `error: "Iterable[Any]" has no attribute "__next__"`

### Expected behavior:
Mypy should give no errors.

### Steps to reproduce/example code:
1. Install mypy and arcade
2. Run mypy on snippet above

## Enhancement request:
Change the type hint on `arcade.sprite_list.SpriteList.__iter__()` from `Iterable[_SpriteType]` to `Iterator[_SpriteType]`.

### What should be added/changed?
See above.

### What would it help with?
It would prevent mypy reporting incorrect type-checking errors.

**Comment by einarf** [comment]
Think this was fixed in experimental branch (2.4)

**Comment by pvcraven** [comment]
Yes, I  think this should be ok in the upcoming 2.4 release.

**Comment by pvcraven** [comment]
Closing, as the fix is part of the 2.4 release.
